### ISSUE TITLE
Simple tap/zoom problem

### ISSUE THREAD

**Issue by pierrese** [dimsemenov/PhotoSwipe#1215]
Hello,

I had a link ("acheter cette photo", disabled by js to open a little window) in the pswp__top-bar and a details/summary element in the caption.

On touch device, when I tap these elements, they open (the window for the a element, the details for the details element) but at the same time it makes a zoom on the photo... There is no problem on computers, just mobile tap devices.

Do you know how I can prevent the zoom when tapping these two elements (I want to keep the normal double-tap zoom) ?

You can see an exemple at https://www.vslonskamalvaud.fr/#&gid=1&pid=501

Thanks for your help !

**Comment by mjau-mjau** [comment]
Because tap/click on these non-anchor elements will "bubble" through to photoswipe, and thus image will zoom, as is default behavior for touch devices.

You could change from:
`<summary class="pswp__summary interactif">Details</summary>`
To:
`<a class="pswp__summary interactif">Details</a>`
This would fix the issue, because a <a> anchor tags in photoswipe do not bubble through the click event.

You could also try the following, but I'm not sure if it will work in your case:
```
$('body').on('click', 'summar.pswp__summary', function(e){
  e.stopImmediatePropagation();
});
```

**Comment by pierrese** [comment]
Hello,
Thank you for your answer. The first solution doesn't work : I have a link in the pswp__top-bar and It also causes a zoom event (that I would also like to eliminate). I tried the second solution, but I didn't notice any change either.

I really don't know what to do...

Thank you for your help,

**Comment by mjau-mjau** [comment]
Sorry, I am not sure what's going on with your links in photoswipe, but seems to be something with your **custom javascript** used for BOTH links. I have a page here, with one normal <a> link and one dynamic <span> (which triggers a JS popup on click), and both work correctly without triggering underlying Photoswipe operations.

The <a> link is already natively ignored by Photoswipe. For the <span> click, we have the following, which does the trick:
```javascript
$('body').on('click', '[data-popup-window]', function(e) {
  e.preventDefault();
  e.stopImmediatePropagation();
  // your code ...
});
```
To conclude, you should be able to use something like the above for your custom javascript click event on the <summary> tag. <a> tags are already ignored. In your case, it could be a problem with the underlying javascript that handles clicks on both these items ... I don't know ...

**Comment by pierrese** [comment]
Thank for the response. It's strange : in fact, when I click on the normal link that are in the `<details>` element, it doesn't cause any problem. So I think that it confirms that the problem is my custom js, but when I add `e.stopImmediatePropagation();` in the code, it doesn't solve the problem...

So my question is : how do you insert your js and html code ? With the functions provided by PhotoSwipe or on your own ?

Thanks,

**Comment by pierrese** [comment]
I use the function isClickableElement as this : 

`isClickableElement: function(el) {
                return el.tagName === 'A' || el.tagName === 'DETAILS';
}`

But with that I stille have my problem.

**Comment by mjau-mjau** [comment]
> I use the function isClickableElement as this :

I checked your code, and the click-event is bound to a `<summary>` tag, not `<details>`, so you would need to instead try:
`isClickableElement: function(el) { return el.tagName === 'A' || el.tagName === 'SUMMARY'; }`

> So my question is : how do you insert your js and html code ? With the functions provided by PhotoSwipe or on your own ?

For our `<span>` click, I don't use anything native to Photoswipe. Just the following, which would need to be bound to your `<summary>` click. I need to see your code in context to see why it might not be working correctly.
```javascript
$('body').on('click', '[data-popup-window]', function(e) {
  e.preventDefault();
  e.stopImmediatePropagation();
  // your code ...
});
```

**Comment by pierrese** [comment]
Thank you. I will try the fist advice (`el.tagName === 'SUMMARY'`) and I will say you if it works. For the moment, I "solved" the problem removing the zoom possibility.
Do you need the js code I use (in a not compressed format) ?

**Comment by slifszyc** [comment]
I am having the same problem and I see [this line](https://github.com/dimsemenov/PhotoSwipe/blob/703c415b1b97e8983fff217644888062f07d308b/src/js/tap.js#L52) in the code which is calling `_shout('doubleTap', p0);` even in single taps.

Why is that? I think that's causing the problem...

**Comment by slifszyc** [comment]
@mjau-mjau why do you say the default behavior for touch devices is zoom? is it intended that way? I think it shouldn't be when `tapToToggleControls` option is set to true for UI Options

**Comment by gosteev** [comment]
this helped me https://github.com/dimsemenov/PhotoSwipe/issues/1198

**Comment by pierrese** [comment]
So... I solved the problem.

First, I tried to put in my init script (in the options)
```
isClickableElement: function(el) {
                return el.tagName === 'A' || el.tagName === 'DETAILS' || el.tagName === 'SUMMARY'; //Pour pouvoir ouvrir le <details> sur les appareils tactiles
            }
```
but it didn't work.

Then, I tried the solution of @mjau-mjau : it works but it also prevent the details of opening... a problem.
```
$('body').on('click', '[data-popup-window]', function(e) {
  e.preventDefault();
  e.stopImmediatePropagation();
  // your code ...
});
```

and at the end I tried the solution given in #1198 (thanks @gosteev), putting of course `if (clickedTagName === "A" || clickedTagName === 'DETAILS' || clickedTagName === 'SUMMARY')` and not just `if (clickedTagName === "A" )`. And it works perfectly.
I also have in my options : 
```
getDoubleTapZoom: function(isMouseClick, item) {
                //return item.initialZoomLevel;
                // isMouseClick          - true if mouse, false if double-tap
			    // item                  - slide object that is zoomed, usually current
			    // item.initialZoomLevel - initial scale ratio of image
			    //                         e.g. if viewport is 700px and image is 1400px,
			    //                              initialZoomLevel will be 0.5
			    if(isMouseClick) {
			        // is mouse click on image or zoom icon
			        // zoom to original
			        return 1;
			        // e.g. for 1400px image:
			        // 0.5 - zooms to 700px
			        // 2   - zooms to 2800px
			
			    } else {
			        // is double-tap
			        // zoom to original if initial zoom is less than 0.7x,
			        // otherwise to 1.5x, to make sure that double-tap gesture always zooms image
			        return item.initialZoomLevel < 0.7 ? 1 : 1.5;
			    }
            }
```

I've just seen the zoom and wide screen toggle buttons disappear on IOS devices (why, it's a mystery) but it's not very important.

**Comment by dimsemenov** [comment]
closed by v5.0
